Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -359,20 +349,27 @@ ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels()
     labelsRegions[label] = shrunkRegion;
   }
 
-  for (InputPixelType label : allLabels)
+  m_Interpolate = false;
+
+  for (SizeValueType i = 0; i < allLabels.size(); i++)
   {
+    const InputPixelType label = allLabels[i];
+    const InputRealType  previousLabel = i > 0 ? allLabels[i - 1] : label - 1;
+    const InputRealType  followingLabel = i < allLabels.size() - 1 ? allLabels[i + 1] : label + 1;
```

Comment:
If `InputPixelType` is `unsigned char` then `label - 1` won't be less than `label` when `label` is 0; likewise, `label + 1` won't be greater than `label` when `label` is 255.  Maybe casting to `InputRealType` earlier would help.  

Even then there is the edge case that `InputPixelType` is `double` and `label` is 1e20; adding or subtracting 1 won't change it because of a lack of precision.  If we permit `label` to be `Inf` or `-Inf` that's trouble too.

I am thinking to not provide `lowerIsovalue` and `upperIsovalue` to `CreateSingleContour`, but instead supply `label`.  Then within `CreateSingleContour` use something like:
```c++
InputPixelType v0 = it.GetPixel(4);
InputPixelType v1 = it.GetPixel(5);
InputPixelType v2 = it.GetPixel(7);
InputPixelType v3 = it.GetPixel(8);
if (m_LabelContours)
{
    v0 = v0 == label;
    v1 = v1 == label;
    v2 = v2 == label;
    v3 = v3 == label;
}
unsigned char  squareCase = 0;
if (v0 > m_contourValue)
{
    squareCase += 1
}
if (v1 > m_contourValue)
{
    squareCase += 2
}
if (v2 > m_contourValue)
{
    squareCase += 4
}
if (v3 > m_contourValue)
{
    squareCase += 8
}
```

Reply:
Please address this in a follow-up patch.

Reply:
If `label` is 0 of type `unsigned char`, label-1 will be an expression of type signed short with value of `-1` due to C++ expression rules, I think. This works for the test, which has 0 and 255 as label values.

Reply:
Updating this logic to cover more cases can be done after we iron out performance issues.

Reply:
I will take a stab at this issue of distinct labels.  @dzenanz